docs/user_guide/storing/docindex.md

```diff
@@ -37,716 +37,75 @@ Currently, DocArray supports the following vector databases:
-which one to use for the search.
-
-The [find()][docarray.index.abstract.BaseDocIndex.find] method returns a named tuple containing the closest
-matching documents and their associated similarity scores.
-
-When searching on subindex level, you can use [find_subindex()][docarray.index.abstract.BaseDocIndex.find_subindex] method, which returns a named tuple containing the subindex documents, similarity scores and their associated root documents.
-
-How these scores are calculated depends on the backend, and can usually be [configured](#customize-configurations).
-
-### Batched search
-
-You can also search for multiple documents at once, in a batch, using the [find_batched()][docarray.index.abstract.BaseDocIndex.find_batched] method.
-
-=== "Search by Documents"
-
-    ```python
-    # create some query Documents
-    queries = DocList[MyDoc](
-        MyDoc(embedding=np.random.rand(128), text=f'query {i}') for i in range(3)
-    )
-
-    # find similar Documents
-    matches, scores = db.find_batched(queries, search_field='embedding', limit=5)
-
-    print(f'{matches=}')
-    print(f'{matches[0].text=}')
-    print(f'{scores=}')
-    ```
-
-=== "Search by raw vectors"
-
-    ```python
-    # create some query vectors
-    query = np.random.rand(3, 128)
-
-    # find similar Documents
-    matches, scores = db.find_batched(query, search_field='embedding', limit=5)
-
-    print(f'{matches=}')
-    print(f'{matches[0].text=}')
-    print(f'{scores=}')
-    ```
-
-The [find_batched()][docarray.index.abstract.BaseDocIndex.find_batched] method returns a named tuple containing
-a list of `DocList`s, one for each query, containing the closest matching documents and their similarity scores.
-
-## Filter search and text search
-
-In addition to vector similarity search, the Document Index interface offers methods for text search and filtered search:
-[text_search()][docarray.index.abstract.BaseDocIndex.text_search] and [filter()][docarray.index.abstract.BaseDocIndex.filter],
-as well as their batched versions [text_search_batched()][docarray.index.abstract.BaseDocIndex.text_search_batched] and [filter_batched()][docarray.index.abstract.BaseDocIndex.filter_batched]. [filter_subindex()][docarray.index.abstract.BaseDocIndex.filter_subindex] is for filter on subindex level.
-
-!!! note
-    The [HnswDocumentIndex][docarray.index.backends.hnswlib.HnswDocumentIndex] implementation does not offer support for filter
-    or text search.
-
-    To see how to perform filter or text search, you can check out other backends that offer support.
-
-## Hybrid search through the query builder
-
-Document Index supports atomic operations for vector similarity search, text search and filter search.
-
-To combine these operations into a single, hybrid search query, you can use the query builder that is accessible
-through [build_query()][docarray.index.abstract.BaseDocIndex.build_query]:
-
+### Initialize the Document Index and add data
 ```python
```

Comment:
can we add 1-2 sentences of explanation to these code snippets? i think they are very self-explanatory, but personally as a user i don't like code snippets without words around them :)